extensions/interactions/GraphInput/directives/GraphInput.js

```diff
@@ -359,9 +408,11 @@ oppia.directive('graphViz', [
             $scope.state.selectedEdge = null;
           };
           $scope.onClickModeButton = function(mode, $event) {
-            $event.preventDefault();
-            $event.stopPropagation();
-            setMode(mode);
+            if ($scope.interactionIsActive) {
+              $event.preventDefault();
```

Comment:
I think you can do the preventDefault() and stopPropagation() in either case.

Reply:
Done